### ISSUE TITLE
Test failures: "Not a HASH reference ..."

### ISSUE THREAD

**Issue by eserte** [Grinnz/Future-Mojo#7]
On some of my smokers the test suite fails:
```
Not a HASH reference at /home/cpansand/.cpan/build/2023122821/Future-Mojo-1.002-0/blib/lib/Future/Mojo.pm line 20.
t/default_loop.t ....... 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
Not a HASH reference at /home/cpansand/.cpan/build/2023122821/Future-Mojo-1.002-0/blib/lib/Future/Mojo.pm line 20.
t/future.t ............. 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
Not a HASH reference at /home/cpansand/.cpan/build/2023122821/Future-Mojo-1.002-0/blib/lib/Future/Mojo.pm line 20.
t/promisify.t .......... 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
```

It seems that this happens if Future::XS is installed.

**Comment by Corion** [comment]
This happens because Future::Mojo reaches deep into the objects instead of using the `->data()` accessor. See PR #6 for the fix.

**Comment by Grinnz** [comment]
This should be fixed in 1.003. Thanks for the report
